forc-pkg/src/pkg.rs

```diff
@@ -1661,17 +1661,25 @@ pub fn check(
         let pkg = &plan.graph[node];
         let path = &plan.path_map[&pkg.id()];
         let manifest = ManifestFile::from_dir(path, sway_git_tag)?;
-        let ast_res = compile_ast(&manifest, &profile, dep_namespace)?;
-        if let CompileAstResult::Success { typed_program, .. } = &ast_res {
-            if let TreeType::Library { .. } = typed_program.kind.tree_type() {
-                namespace_map.insert(node, typed_program.root.namespace.clone());
+
+        let source = manifest.entry_string()?;
+        let sway_build_config =
+            sway_build_config(manifest.dir(), &manifest.entry_path(), &profile)?;
+        let parsed_result = sway_core::parse(source, Some(&sway_build_config));
```

Comment:
Maybe we should refactor these 4 lines into a `parse` function that we can publicly export from `forc_pkg`?

Something like:

```rust
pub fn parse(manifest: &ManifestFile, silent: bool) -> CompileResult<ParseProgram> { ... }
```
^ I can imagine this might be a useful short-hand for the formatter too.